MapboxVisionAR/VisionARManager.swift

```diff
@@ -11,42 +11,35 @@ import MapboxVisionCore
 import MapboxVisionARCore
 import MapboxVision
 
-public final class VisionARManager: VisionARDelegateNative {
+public final class VisionARManager {
     
     private var native: VisionARManagerNative?
-    private let delegate: VisionARDelegate
+    private var delegate: VisionARDelegate?
     
-    public static func create(visionManager: VisionManager, delegate: VisionARDelegate) -> VisionARManager {
-        let mananger = VisionARManager(delegate)
-        mananger.native = VisionARManagerNative.create(withVisionManager: visionManager.native, delegate: mananger)
-        return mananger
-    }
-    
-    private init(_ delegate: VisionARDelegate) {
-        self.delegate = delegate
-    }
-    
-    public func setRoute(_ route: Route) {
-        guard let native = native else {
-            assertionFailure()
-            return
-        }
-        native.setRoute(route)
+    public static func create(visionManager: VisionManager, delegate: VisionARDelegate? = nil) -> VisionARManager {
+        let manager = VisionARManager()
+        manager.native = VisionARManagerNative.create(visionManager: visionManager.native, delegate: manager)
+        manager.delegate = delegate
+        return manager
     }
     
     public func destroy() {
-        guard let native = native else {
```

Comment:
Why you remove this check?

Reply:
Didn't really get its purpose. Was it needed to guard from incorrect configuration in `create`?

Reply:
I think we need to know (by assert) what something is going wrong. For example: We trying to call 'setRoute' after 'destroy'.

Reply:
Now that `native` is being nulled in `destroy`, we may check for multiple `destroy` calls to warn user about incorrect lifecycle management.
On the other hand, setter methods on "destroyed" manager producing no-op, seems fine to me.